ddtrace/tracer.py

```diff
@@ -312,6 +313,12 @@ def deregister_on_start_span(self, func):
     def debug_logging(self):
         return log.isEnabledFor(logging.DEBUG)
 
+    def global_excepthook(self, tp, value, traceback):
+        """The global tracer except hook."""
+        if asbool(os.getenv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", True)):
+            telemetry_writer.add_error(1, tp)
+            telemetry_writer.enable()
```

Comment:
We now attempt to enable the telemetry_writer in few places (ddtrace/internal/writer.py, ddtrace/__init__.py). Can we do this one place? Ideally in ddtrace/__init__.py

Reply:
We need to enable it here, because otherwise it's not started early enough to write the error before the app closes out.

Reply:
I see. Previously we queued app started events when the writer was enabled. Now we have to wait until we collect all possible start up errors before sending this event. This can be tricky.

I am not sure if there is an easy way to determine if an application has "successfully started".

Suggestion: We submit an app-started event after the first trace is sent to the agent. Any errors queued after the first trace is submitted will not be considered a start up error and will be ignored by the telemetry writer. The con of this approach is that we need to delay sending all telemetry events until a trace is sent. We can not send metrics/logs/integrations/dependencies/etc until the `app-started` event is ready. This approach could result in a memory leak. A workaround could be to set a max size on the telemetry queue but then we will drop metrics/logs/....

Thoughts? @Kyle-Verhoog

Reply:
I discussed this issue offline with Mehul. 

1. We will continue to send `app-started` after the first trace is sent to the agent. This will allow us to collect errors associated with connection to the agent. 
2. We can send telemetry logs/metrics as they are queued. There is no need to block until app-started is sent. This will resolve the memory leak issue. Metrics/Log events follow a special pathway and are never mapped to "lifecycle" events (ex: app started/app closed/...).